### PR TITLE
Scenes: Add 'Import from library' functionality

### PR DESCRIPTION
**Pull request by mdvictor**
Migrates the `Import from library` functionality in scenes. Part of https://github.com/grafana/grafana/issues/82316

Depends on https://github.com/grafana/grafana/pull/83231

### CODE REVIEW

**Review by harisrozajac — COMMENTED** [review]
This looks good to me! Nice work! Should we add this functionality to the empty state? Here: https://github.com/grafana/grafana/blob/70009201d44c2d0ab39cc77081808a69a6c4fd63/public/app/features/dashboard/dashgrid/DashboardEmpty.tsx#L116

**Comment by mdvictor** [comment]
> This looks good to me! Nice work! Should we add this functionality to the empty state? Here:
> 
> https://github.com/grafana/grafana/blob/70009201d44c2d0ab39cc77081808a69a6c4fd63/public/app/features/dashboard/dashgrid/DashboardEmpty.tsx#L116

Working on that [here](https://github.com/grafana/grafana/pull/83522). There's some compatibility layer issue that needs to be addressed, that's why I kept it separate.

File: public/app/features/dashboard-scene/scene/AddLibraryPanelWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      throw new Error('Trying to add a library panel in a layout that is not SceneGridLayout');
+    }
+
+    const sceneGridLayout = this._dashboard.state.body;
+
+    const body = new LibraryVizPanel({
+      title: 'Panel Title',
+      uid: panelInfo.uid,
+      name: panelInfo.name,
+      //TODO modify this to be panelKey after https://github.com/grafana/grafana/pull/83420 gets merged
```
Comment (oscarkilhed) [inline public/app/features/dashboard-scene/scene/AddLibraryPanelWidget.tsx:88]: This has been merged now :)

Comment (mdvictor) [reply public/app/features/dashboard-scene/scene/AddLibraryPanelWidget.tsx:88]: Right on, thank you! Will push a change asap

File: public/app/features/dashboard-scene/scene/AddLibraryPanelWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      key: this.state.key,
+    });
+
+    for (const child of sceneGridLayout.state.children) {
+      if (child instanceof SceneGridItem && child.state.key === this.parent?.state.key) {
+        child.setState({ body });
+        return;
```
Comment (oscarkilhed) [inline public/app/features/dashboard-scene/scene/AddLibraryPanelWidget.tsx:95]: Do we really need this search? Isn't this.parent the same as child? Then meybe we can do `this.parent.setState({body})` ?

Could be that I'm missing something here.

Comment (mdvictor) [reply public/app/features/dashboard-scene/scene/AddLibraryPanelWidget.tsx:95]: That's true! Thanks for catching that, makes the code so much better. Pushing a commit soon!

**Review by oscarkilhed — APPROVED** [review]
Tested it out and I think it looks good!

**Review by ivanortegaalba — APPROVED** [review]
Great tests @mdvictor! 💯

**Pull request merged**